dbms/src/Flash/tests/gtest_expand_executor.cpp

```diff
@@ -0,0 +1,440 @@
+    ///  count(s1)   s2      groupingID  *    s2                              |
+    ///   NULL     "apple"      2           "apple"       // join             |
+    ///   NULL     "banana"     2            NULL                             |
+    ///                                     "banana"                          +------------->  Child DAG Query Block
+    ///                                                                       |
+    ///   NULL     "apple"      2          "apple"                            |
+    ///   NULL     "banana"     2          "banana"                           |
+    ///          |                                             ---------------+
+    ///          v                                                            |
+    ///  count(s1)  groupingID             // projection                      |
+    ///   NULL         2                                                      |
+    ///   NULL         2                                                      |
+    ///          |                                                            +------------->  Parent DAG Query Block
+    ///          v                                                            |
+    ///   count(s1)  groupingID            // topN                            |
+    ///   NULL         2                                                      |
+    ///   NULL         2                                                      |
+    ///                                                        ---------------+
+    ///
+    executeAndAssertColumnsEqual(
+        request,
+        {
+            toNullableVec<UInt64>({{}, {}}),
+            toVec<UInt64>({2, 2}),
+        });
+
+
+    /// assert the input stream plan format. (under planner-enabled mode)
+    String expected = R"(
+CreatingSets
+ Union: <for join>
+  HashJoinBuild x 10: <join build, build_side_root_executor_id = project_4>, join_kind = Inner
+   Expression: <append join key and join filters for build side>
+    Expression: <final projection>
+     Expression: <projection>
+      MockTableScan
+ Union: <for test>
+  Expression x 10: <final projection>
+   SharedQuery: <restore concurrency>
+    MergeSorting, limit = 2
+     Union: <for partial order>
+      PartialSorting x 10: limit = 2
+       Expression: <projection>
+        Expression: <remove useless column after join>
+         HashJoinProbe: <join probe, join_executor_id = Join_5, has_non_joined_data = false>
+          Expression: <final projection>
+           Expression: <expand, expand_executor_id = expand_2: grouping set [{<count(exchange_receiver_0)_collator_46 >}{<any(exchange_receiver_1)_collator_46 >}]>
+            Expression: <expr after aggregation>
+             SharedQuery: <restore concurrency>
+              ParallelAggregating, max_threads: 10, final: true
+               MockExchangeReceiver x 10)";
+    ASSERT_BLOCKINPUTSTREAM_EQAUL(expected, request, 10);
```

Comment:
better to move to `gtest_interpreter.cpp`, `gtest_planner_interpreter.cpp` and `gtest_pipeline_interpreter.cpp`...